src/pages/de.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * A backwards compatibility page for redirecting legacy links
+ * to their proper current locations.
+ *
+ * @author psarando
+ */
+import React from "react";
+
+import { useRouter } from "next/router";
+
+import {
+    getAppLaunchPath,
+    getListingPath,
+    getQuickLaunchPath,
+} from "components/apps/utils";
+import { getFolderPage } from "components/data/utils";
+import systemId from "components/models/systemId";
+import BasicTable from "components/utils/BasicTable";
+
+export default function DeLegacyRedirector() {
+    const router = useRouter();
+    const query = router.query;
+
+    let redirectPath;
+    switch (query?.type) {
+        case "apps":
+            if (query["app-category"]) {
+                redirectPath = getListingPath(
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    JSON.stringify({
+                        system_id: query["system-id"] || systemId.de,
+                        id: query["app-category"],
+                    })
+                );
```

Comment:
Even though this returns a `{pathname, query}` object, it turns out `router.replace` won't auto-stringify the `selectedCategory` query param's value 😞